### ISSUE TITLE
Support for multiple responses

### ISSUE THREAD

**Issue by tomcully** [krisboit/protractor-xmlhttprequest-mock#17]
It would be nice to be able to provide a list of responses for a given path so that you can call the same endpoint many times and have different results.

It would allow you to mock an e2e flow

e.g.
```
MockService.addMock('getEventsMock', {
    path: '/some/api',
    response: [    // Config becomes an array
        {
            status: 200,
            date: { "name": "first event", "hasBeenCalled": false }
        },
        {
            status: 200,
            date: { "name": "first-event", "hasBeenCalled": true }
        }
    ]
});
```

I'll see if I can put a pr together for this. maybe a separate add function like `addMocks` might work better

**Comment by krisboit** [comment]
@tomcully, man sorry for the late responses, i was a little busy lately. Also thanks for contributing.

I like your suggestion. Just keep in mind that we should keep the function signature backward compatible, so i would suggest `response` to be `Response | Response[]`, where `Response` is an interface.

**Comment by tomcully** [comment]
@krisboit no problems! 

Cool, i agreed on backward compatibility. I’ll knock something together and see what you think 👍🏻

**Comment by krisboit** [comment]
solved in 1.4.0
